File: project/backend-services/smart-irrigation-backend/infrastructure/endpoint/graphql/src/main/java/pt/sensae/services/smart/irrigation/backend/infrastructure/endpoint/graphql/model/data/SensorDataHistory.java
package pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.data;

import pt.sensae.services.smart.irrigation.backend.application.model.data.SensorDataHistoryDTO;
import pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.device.DeviceLedgerHistoryEntryDTOImpl;
import pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.device.DeviceTypeDTOImpl;

import java.util.Set;
import java.util.UUID;

public record SensorDataHistory(UUID id, DeviceTypeDTOImpl type,
                                Set<DeviceLedgerHistoryEntryDTOImpl> ledger) implements SensorDataHistoryDTO {
}

File: project/frontend-services/libs/smart-irrigation/model/src/lib/model/history/SensorDataHistory.ts
import {DeviceType} from "../device/DeviceType";
import {DeviceLedgerHistoryEntry} from "./DeviceLedgerHistoryEntry";

export class SensorDataHistory {
  constructor(public id: string, public type: DeviceType, public ledger: Array<DeviceLedgerHistoryEntry>) {
  }
}
